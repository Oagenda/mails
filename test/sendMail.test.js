const _ = require( 'lodash' );
const nodemailer = require( 'nodemailer' );
const mails = require( '../index' );
const config = require( '../config' );
const makeLabelGetter = require( '../utils/makeLabelGetter' );

let account;
const getEtherealTransport = () => ( {
  host: 'smtp.ethereal.email',
  port: 587,
  secure: false, // true for 465, false for other ports
  auth: {
    user: account.user, // generated ethereal user
    pass: account.pass // generated ethereal password
  }
} );

beforeAll( async () => {
  account = await nodemailer.createTestAccount();
} );

describe( 'sendMail', () => {
  describe( 'JSON transport', () => {
    beforeAll( async () => {
      await config.init( {
        transport: {
          jsonTransport: true
        },
        defaults: {
          data: {
            domain: 'https://openagenda.com'
          }
        },
        queueName: 'mailsTest-sendMail'
      } );

      await config.queue.clear();
    } );

    it( 'send email - helloWorld template', async () => {
      const { results, errors } = await mails( {
        template: 'helloWorld',
        data: {
          username: 'bertho'
        },
        to: [
          'kevin.bertho@gmail.com, kevin.berthommier@openagenda.com',
          { address: '"Kaoré" <kaore@openagenda.com>', data: { username: 'kaore' } }
        ]
      } );

      expect( results ).toHaveLength( 3 );
      expect( errors ).toHaveLength( 0 );

      expect( await config.queue.waitAndPop() ).toMatchSnapshot( 'kevin.bertho@gmail.com mail' );
      expect( await config.queue.waitAndPop() ).toMatchSnapshot( 'kevin.berthommier@openagenda.com mail' );
      expect( await config.queue.waitAndPop() ).toMatchSnapshot( 'kaore@openagenda.com mail' );
    } );

    it( 'send a mail to an invalid email returns with errors', async () => {
      const { errors } = await mails( {
        template: 'helloWorld',
        data: {
          username: 'bertho'
        },
        to: 'kevin.bertho@@gmail',
        queue: false
      } );

      expect( errors ).toMatchSnapshot();
    } );

    it( 'sendMail can not override the defaults', async () => {
      const { results } = await mails( {
        template: 'helloWorld',
        data: {
          domain: 'https://google.com',
          username: 'bertho'
        },
        to: {
          address: 'kevin.bertho@gmail.com',
          data: {
            domain: 'https://bertho.io'
          }
        },
        queue: false
      } );

      const message = JSON.parse( results[ 0 ].message );

      expect( message.data.domain ).toBe( 'https://openagenda.com' );
    } );
  } );

  describe( 'Euthreal transport', () => {
    beforeAll( async () => {
      await config.init( {
        transport: getEtherealTransport(),
        defaults: {
          from: 'no-reply@openagenda.com',
          data: {
            domain: 'https://openagenda.com'
          }
        },
        queueName: 'mailsTest-sendMail'
      } );

      await config.queue.clear();
    } );

    it( 'send email directly (without queue) - helloWorld template', async () => {
      const { results, errors } = await mails( {
        template: 'helloWorld',
        data: {
          username: 'Bertho'
        },
        to: [
          '"Kévin Berthommier" <kevin.bertho@gmail.com>, kevin.berthommier@openagenda.com',
          { address: 'kaore@openagenda.com', data: { username: 'Kaore' } }
        ],
        queue: false
      } );

      expect( errors ).toHaveLength( 0 );

      for ( const result of results ) {
        console.log( `Preview URL: ${nodemailer.getTestMessageUrl( result )}` );
        expect( result.response ).toContain( '250' );
      }

      expect( results.map( v => _.omit( v, 'envelopeTime', 'messageId', 'messageTime', 'response' ) ) ).toMatchSnapshot();
    } );

    it( 'override default "from" value', async () => {
      const from = 'admin@openagenda.com';

      const { results, errors } = await mails( {
        template: 'helloWorld',
        from,
        to: [
          {
            address: 'kevin.bertho@gmail.com',
            name: 'Kévin Berthommier',
            data: { username: 'Bertho' }
          }
        ],
        queue: false
      } );

      expect( errors ).toHaveLength( 0 );

      for ( const result of results ) {
        console.log( `Preview URL: ${nodemailer.getTestMessageUrl( result )}` );
        expect( result.response ).toContain( '250' );
      }

      expect( results[ 0 ].envelope.from ).toBe( from );

      expect( results.map( v => _.omit( v, 'envelopeTime', 'messageId', 'messageTime', 'response' ) ) ).toMatchSnapshot();
    } );
  } );

  describe( 'translations', () => {
    it( 'take a default lang', async () => {
      await config.init( {
        transport: getEtherealTransport(),
        defaults: {
          data: {
            domain: 'https://openagenda.com'
          },
          lang: 'en'
        },
        translations: {
          labels: {
            'helloWorld-i18n': {
              hello: {
                fr: 'Salut %username%',
                en: 'Hello %username%'
              },
              goToOA: {
                fr: 'Aller sur OpenAgenda',
                en: 'Go to OpenAgenda'
              }
            }
          },
          makeLabelGetter
        },
        queueName: 'mailsTest-sendMail'
      } );

      const { results, errors } = await mails( {
        template: 'helloWorld-i18n',
        to: {
          address: 'kevin.bertho@gmail.com',
          data: { username: 'bertho' }
        },
        queue: false
      } );

      expect( errors ).toHaveLength( 0 );
      expect( results ).toHaveLength( 1 );

      for ( const result of results ) {
        console.log( `Preview URL: ${nodemailer.getTestMessageUrl( result )}` );
        expect( result.response ).toContain( '250' );
      }

      expect( results.map( v => _.omit( v, 'envelopeTime', 'messageId', 'messageTime', 'response' ) ) ).toMatchSnapshot();
    } );

    it( 'choose a lang per recipient', async () => {
      await config.init( {
        transport: getEtherealTransport(),
        defaults: {
          data: {
            domain: 'https://openagenda.com'
          },
          lang: 'en'
        },
        translations: {
          labels: {
            'helloWorld-i18n': {
              hello: {
                fr: 'Salut %username%',
                en: 'Hello %username%'
              },
              goToOA: {
                fr: 'Aller sur OpenAgenda',
                en: 'Go to OpenAgenda'
              }
            }
          },
          makeLabelGetter
        },
        queueName: 'mailsTest-sendMail'
      } );

      const { results, errors } = await mails( {
        template: 'helloWorld-i18n',
        to: {
          address: 'kevin.bertho@gmail.com',
          data: { username: 'bertho' },
          lang: 'fr'
        },
        queue: false
      } );

      expect( errors ).toHaveLength( 0 );
      expect( results ).toHaveLength( 1 );

      for ( const result of results ) {
        console.log( `Preview URL: ${nodemailer.getTestMessageUrl( result )}` );
        expect( result.response ).toContain( '250' );
      }

      expect( results.map( v => _.omit( v, 'envelopeTime', 'messageId', 'messageTime', 'response' ) ) ).toMatchSnapshot();
    } );
  } );
} );
