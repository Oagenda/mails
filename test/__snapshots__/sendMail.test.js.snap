// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sendMail Euthreal transport override default "from" value 1`] = `
Array [
  Object {
    "accepted": Array [
      "kevin.bertho@gmail.com",
    ],
    "envelope": Object {
      "from": "admin@openagenda.com",
      "to": Array [
        "kevin.bertho@gmail.com",
      ],
    },
    "messageSize": 6488,
    "rejected": Array [],
  },
]
`;

exports[`sendMail Euthreal transport send email directly (without queue) - helloWorld template 1`] = `
Array [
  Object {
    "accepted": Array [
      "kevin.bertho@gmail.com",
    ],
    "envelope": Object {
      "from": "no-reply@openagenda.com",
      "to": Array [
        "kevin.bertho@gmail.com",
      ],
    },
    "messageSize": 6528,
    "rejected": Array [],
  },
  Object {
    "accepted": Array [
      "kevin.berthommier@openagenda.com",
    ],
    "envelope": Object {
      "from": "no-reply@openagenda.com",
      "to": Array [
        "kevin.berthommier@openagenda.com",
      ],
    },
    "messageSize": 6501,
    "rejected": Array [],
  },
  Object {
    "accepted": Array [
      "kaore@openagenda.com",
    ],
    "envelope": Object {
      "from": "no-reply@openagenda.com",
      "to": Array [
        "kaore@openagenda.com",
      ],
    },
    "messageSize": 6486,
    "rejected": Array [],
  },
]
`;

exports[`sendMail JSON transport send a mail to an invalid email returns with errors 1`] = `
Array [
  [VError: Invalid email address],
]
`;

exports[`sendMail JSON transport send email - helloWorld template: kaore@openagenda.com mail 1`] = `
Object {
  "data": Object {
    "domain": "https://openagenda.com",
    "username": "kaore",
  },
  "html": "
    <!doctype html>
    <html xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\">
      <head>
        <title>
          
        </title>
        <!--[if !mso]><!-- -->
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
        <!--<![endif]-->
        <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <style type=\\"text/css\\">
          #outlook a { padding:0; }
          .ReadMsgBody { width:100%; }
          .ExternalClass { width:100%; }
          .ExternalClass * { line-height:100%; }
          body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }
          table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }
          img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }
          p { display:block;margin:13px 0; }
        </style>
        <!--[if !mso]><!-->
        <style type=\\"text/css\\">
          @media only screen and (max-width:480px) {
            @-ms-viewport { width:320px; }
            @viewport { width:320px; }
          }
        </style>
        <!--<![endif]-->
        <!--[if mso]>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        <![endif]-->
        <!--[if lte mso 11]>
        <style type=\\"text/css\\">
          .outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->
        
      <!--[if !mso]><!-->
        <link href=\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\" rel=\\"stylesheet\\" type=\\"text/css\\">
        <style type=\\"text/css\\">
          @import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);
        </style>
      <!--<![endif]-->

    
        
    <style type=\\"text/css\\">
      @media only screen and (min-width:480px) {
        .mj-column-per-100 { width:100% !important; max-width: 100%; }
      }
    </style>
    
  
        <style type=\\"text/css\\">
        
        
        </style>
        
      </head>
      <body>
        
        
      <div
         style=\\"\\"
      >
        
      
      <!--[if mso | IE]>
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" style=\\"width:600px;\\" width=\\"600\\"
      >
        <tr>
          <td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\">
      <![endif]-->
    
      
      <div  style=\\"Margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:20px 0;text-align:center;vertical-align:top;\\"
              >
                <!--[if mso | IE]>
                  <table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
                
        <tr>
      
            <td
               class=\\"\\" style=\\"vertical-align:top;width:600px;\\"
            >
          <![endif]-->
            
      <div
         class=\\"mj-column-per-100 outlook-group-fix\\" style=\\"font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        
            <tr>
              <td
                 align=\\"left\\" style=\\"font-size:0px;padding:10px 25px;word-break:break-word;\\"
              >
                
      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;line-height:1;text-align:left;color:#000000;\\"
      >
        Hello kaore !
      </div>
    
              </td>
            </tr>
          
            <tr>
              <td
                 align=\\"center\\" vertical-align=\\"middle\\" style=\\"font-size:0px;padding:10px 25px;word-break:break-word;\\"
              >
                
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:separate;line-height:100%;\\"
      >
        <tr>
          <td
             align=\\"center\\" bgcolor=\\"#2494c7\\" role=\\"presentation\\" style=\\"border:none;border-radius:5px;cursor:auto;padding:10px 25px;\\" valign=\\"middle\\"
          >
            <a
               href=\\"https://openagenda.com\\" style=\\"background:#2494c7;color:white;font-family:Helvetica;font-size:13px;font-weight:normal;line-height:120%;Margin:0;text-decoration:none;text-transform:none;\\" target=\\"_blank\\"
            >
              Go to OpenAgenda
            </a>
          </td>
        </tr>
      </table>
    
              </td>
            </tr>
          
      </table>
    
      </div>
    
          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      <![endif]-->
    
    
      </div>
    
      </body>
    </html>
  ",
  "subject": "Hello kaore !
",
  "template": "helloWorld",
  "text": "Hello kaore !
Go to https://openagenda.com
",
  "to": Object {
    "address": "kaore@openagenda.com",
    "data": Object {
      "username": "kaore",
    },
    "name": "Kaor√©",
  },
}
`;

exports[`sendMail JSON transport send email - helloWorld template: kevin.bertho@gmail.com mail 1`] = `
Object {
  "data": Object {
    "domain": "https://openagenda.com",
    "username": "bertho",
  },
  "html": "
    <!doctype html>
    <html xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\">
      <head>
        <title>
          
        </title>
        <!--[if !mso]><!-- -->
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
        <!--<![endif]-->
        <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <style type=\\"text/css\\">
          #outlook a { padding:0; }
          .ReadMsgBody { width:100%; }
          .ExternalClass { width:100%; }
          .ExternalClass * { line-height:100%; }
          body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }
          table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }
          img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }
          p { display:block;margin:13px 0; }
        </style>
        <!--[if !mso]><!-->
        <style type=\\"text/css\\">
          @media only screen and (max-width:480px) {
            @-ms-viewport { width:320px; }
            @viewport { width:320px; }
          }
        </style>
        <!--<![endif]-->
        <!--[if mso]>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        <![endif]-->
        <!--[if lte mso 11]>
        <style type=\\"text/css\\">
          .outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->
        
      <!--[if !mso]><!-->
        <link href=\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\" rel=\\"stylesheet\\" type=\\"text/css\\">
        <style type=\\"text/css\\">
          @import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);
        </style>
      <!--<![endif]-->

    
        
    <style type=\\"text/css\\">
      @media only screen and (min-width:480px) {
        .mj-column-per-100 { width:100% !important; max-width: 100%; }
      }
    </style>
    
  
        <style type=\\"text/css\\">
        
        
        </style>
        
      </head>
      <body>
        
        
      <div
         style=\\"\\"
      >
        
      
      <!--[if mso | IE]>
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" style=\\"width:600px;\\" width=\\"600\\"
      >
        <tr>
          <td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\">
      <![endif]-->
    
      
      <div  style=\\"Margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:20px 0;text-align:center;vertical-align:top;\\"
              >
                <!--[if mso | IE]>
                  <table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
                
        <tr>
      
            <td
               class=\\"\\" style=\\"vertical-align:top;width:600px;\\"
            >
          <![endif]-->
            
      <div
         class=\\"mj-column-per-100 outlook-group-fix\\" style=\\"font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        
            <tr>
              <td
                 align=\\"left\\" style=\\"font-size:0px;padding:10px 25px;word-break:break-word;\\"
              >
                
      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;line-height:1;text-align:left;color:#000000;\\"
      >
        Hello bertho !
      </div>
    
              </td>
            </tr>
          
            <tr>
              <td
                 align=\\"center\\" vertical-align=\\"middle\\" style=\\"font-size:0px;padding:10px 25px;word-break:break-word;\\"
              >
                
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:separate;line-height:100%;\\"
      >
        <tr>
          <td
             align=\\"center\\" bgcolor=\\"#2494c7\\" role=\\"presentation\\" style=\\"border:none;border-radius:5px;cursor:auto;padding:10px 25px;\\" valign=\\"middle\\"
          >
            <a
               href=\\"https://openagenda.com\\" style=\\"background:#2494c7;color:white;font-family:Helvetica;font-size:13px;font-weight:normal;line-height:120%;Margin:0;text-decoration:none;text-transform:none;\\" target=\\"_blank\\"
            >
              Go to OpenAgenda
            </a>
          </td>
        </tr>
      </table>
    
              </td>
            </tr>
          
      </table>
    
      </div>
    
          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      <![endif]-->
    
    
      </div>
    
      </body>
    </html>
  ",
  "subject": "Hello bertho !
",
  "template": "helloWorld",
  "text": "Hello bertho !
Go to https://openagenda.com
",
  "to": Object {
    "address": "kevin.bertho@gmail.com",
    "name": "",
  },
}
`;

exports[`sendMail JSON transport send email - helloWorld template: kevin.berthommier@openagenda.com mail 1`] = `
Object {
  "data": Object {
    "domain": "https://openagenda.com",
    "username": "bertho",
  },
  "html": "
    <!doctype html>
    <html xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\">
      <head>
        <title>
          
        </title>
        <!--[if !mso]><!-- -->
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
        <!--<![endif]-->
        <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <style type=\\"text/css\\">
          #outlook a { padding:0; }
          .ReadMsgBody { width:100%; }
          .ExternalClass { width:100%; }
          .ExternalClass * { line-height:100%; }
          body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }
          table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }
          img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }
          p { display:block;margin:13px 0; }
        </style>
        <!--[if !mso]><!-->
        <style type=\\"text/css\\">
          @media only screen and (max-width:480px) {
            @-ms-viewport { width:320px; }
            @viewport { width:320px; }
          }
        </style>
        <!--<![endif]-->
        <!--[if mso]>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        <![endif]-->
        <!--[if lte mso 11]>
        <style type=\\"text/css\\">
          .outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->
        
      <!--[if !mso]><!-->
        <link href=\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\" rel=\\"stylesheet\\" type=\\"text/css\\">
        <style type=\\"text/css\\">
          @import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);
        </style>
      <!--<![endif]-->

    
        
    <style type=\\"text/css\\">
      @media only screen and (min-width:480px) {
        .mj-column-per-100 { width:100% !important; max-width: 100%; }
      }
    </style>
    
  
        <style type=\\"text/css\\">
        
        
        </style>
        
      </head>
      <body>
        
        
      <div
         style=\\"\\"
      >
        
      
      <!--[if mso | IE]>
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" style=\\"width:600px;\\" width=\\"600\\"
      >
        <tr>
          <td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\">
      <![endif]-->
    
      
      <div  style=\\"Margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:20px 0;text-align:center;vertical-align:top;\\"
              >
                <!--[if mso | IE]>
                  <table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
                
        <tr>
      
            <td
               class=\\"\\" style=\\"vertical-align:top;width:600px;\\"
            >
          <![endif]-->
            
      <div
         class=\\"mj-column-per-100 outlook-group-fix\\" style=\\"font-size:13px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        
            <tr>
              <td
                 align=\\"left\\" style=\\"font-size:0px;padding:10px 25px;word-break:break-word;\\"
              >
                
      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;line-height:1;text-align:left;color:#000000;\\"
      >
        Hello bertho !
      </div>
    
              </td>
            </tr>
          
            <tr>
              <td
                 align=\\"center\\" vertical-align=\\"middle\\" style=\\"font-size:0px;padding:10px 25px;word-break:break-word;\\"
              >
                
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:separate;line-height:100%;\\"
      >
        <tr>
          <td
             align=\\"center\\" bgcolor=\\"#2494c7\\" role=\\"presentation\\" style=\\"border:none;border-radius:5px;cursor:auto;padding:10px 25px;\\" valign=\\"middle\\"
          >
            <a
               href=\\"https://openagenda.com\\" style=\\"background:#2494c7;color:white;font-family:Helvetica;font-size:13px;font-weight:normal;line-height:120%;Margin:0;text-decoration:none;text-transform:none;\\" target=\\"_blank\\"
            >
              Go to OpenAgenda
            </a>
          </td>
        </tr>
      </table>
    
              </td>
            </tr>
          
      </table>
    
      </div>
    
          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      <![endif]-->
    
    
      </div>
    
      </body>
    </html>
  ",
  "subject": "Hello bertho !
",
  "template": "helloWorld",
  "text": "Hello bertho !
Go to https://openagenda.com
",
  "to": Object {
    "address": "kevin.berthommier@openagenda.com",
    "name": "",
  },
}
`;

exports[`sendMail translations choose a lang per recipient 1`] = `
Array [
  Object {
    "accepted": Array [
      "kevin.bertho@gmail.com",
    ],
    "envelope": Object {
      "from": "",
      "to": Array [
        "kevin.bertho@gmail.com",
      ],
    },
    "messageSize": 6473,
    "rejected": Array [],
  },
]
`;

exports[`sendMail translations take a default lang 1`] = `
Array [
  Object {
    "accepted": Array [
      "kevin.bertho@gmail.com",
    ],
    "envelope": Object {
      "from": "",
      "to": Array [
        "kevin.bertho@gmail.com",
      ],
    },
    "messageSize": 6465,
    "rejected": Array [],
  },
]
`;
